pkgs/applications/science/logic/coq/default.nix

```diff
@@ -109,15 +109,18 @@ self = stdenv.mkDerivation {
   nativeBuildInputs = [ pkgconfig ]
   ++ stdenv.lib.optional (!versionAtLeast "8.6") gnumake42
   ;
-  buildInputs = [ ncurses ocamlPackages.ocaml ocamlPackages.findlib ]
+  buildInputs = [ ncurses ]
   ++ stdenv.lib.optional (!versionAtLeast "8.10") ocamlPackages.camlp5
   ++ stdenv.lib.optional (!versionAtLeast "8.12") ocamlPackages.num
   ++ stdenv.lib.optionals buildIde
     (if versionAtLeast "8.10"
      then [ ocamlPackages.lablgtk3-sourceview3 glib gnome3.defaultIconTheme wrapGAppsHook ]
      else [ ocamlPackages.lablgtk ]);
 
-  propagatedBuildInputs = stdenv.lib.optional (versionAtLeast "8.12") ocamlPackages.num;
+  propagatedBuildInputs = [ ocamlPackages.ocaml ocamlPackages.findlib ]
+    ++ stdenv.lib.optional (versionAtLeast "8.12") ocamlPackages.num;
+
+  propagatedUserEnvPkgs = [ ocamlPackages.ocaml ocamlPackages.findlib ];
```

Comment:
Where is this attribute documented? What’s the relation to `propagatedBuildInputs`? Why is its value different?

Reply:
I didn't know about this attribute, it is a suggestion from @dtzWill in #34657. I also needed to extend `propagatedBuildInput` to resolve the issue inside a `nix-shell -p coq` and not just when people install Coq. I don't know if there will be limitations due to `num` not being installed but it is well known that Nix-installed libraries do not work outside a `nix-shell` anyway, so there is really no reason to include `num` in `propagatedUserEnvPkgs`.